src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_validators.py

```diff
@@ -123,3 +123,16 @@ def validate_subnet(cmd, namespace):
     else:
         raise CLIError('incorrect usage: [--subnet ID | --subnet NAME --vnet-name NAME]')
     delattr(namespace, 'vnet_name')
+
+
+###############################################
+#                sql managed instance         #
+###############################################
+
+
+def validate_managed_instance_storage_size(cmd, namespace):
```

Comment:
Why validate on client side? There is already validation on server side, and perhaps in the future more storage options will be available

Reply:
Unfortunately, the error when customer specifies wrong size is not well propagated back and customer cannot see actual reason for failure. We have a task to fix this, but in the meantime I was thinking to prevent this issue by validating it on client size.